src/program/program.py

```diff
@@ -212,39 +212,43 @@ def _retry_library(self) -> None:
             for item_id in result.scalars():
                 self.em.add_event(Event(emitted_by="RetryLibrary", item_id=item_id))
 
-    def _reindex_ongoing(self) -> None:
-        """Reindex ongoing items."""
+    def _update_ongoing(self) -> None:
+        """Update state for ongoing and unreleased items."""
         with db.Session() as session:
-            results = session.execute(
+            item_ids = session.execute(
                 select(MediaItem.id)
                 .where(MediaItem.type.in_(["movie", "show"]))
                 .where(MediaItem.last_state.in_([States.Ongoing, States.Unreleased]))
-                .where(or_(MediaItem.aired_at <= datetime.now(), MediaItem.aired_at.is_(None)))
             ).scalars().all()
 
-            if len(results) == 0:
-                logger.log("PROGRAM", "No ongoing or unreleased items to reindex.")
+            if not item_ids:
+                logger.log("PROGRAM", "No ongoing or unreleased items to update.")
                 return
 
-            logger.log("PROGRAM", f"Reindexing {len(results)} unreleased and ongoing items.")
+            logger.debug(f"Updating state for {len(item_ids)} ongoing and unreleased items.")
 
-            for item_id in results:
+            counter = 0
+            for item_id in item_ids:
                 try:
                     item = session.execute(select(MediaItem).filter_by(id=item_id)).unique().scalar_one_or_none()
                     if item:
-                        for indexed_item in TraktIndexer().run(item):
-                            indexed_item.store_state()
-                            session.merge(indexed_item)
-                            logger.debug(f"Reindexed {indexed_item.log_string} ({indexed_item.id})")
+                        previous_state = item.last_state
+                        item.store_state()
```

Comment:
Could store_state() return previous, new state tuple instead?

Reply:
that could be useful